### DISCUSSION TITLE
How does fastapi-user handle email verifications ?

### DISCUSSION THREAD

**Discussion by striker561** [fastapi-users/fastapi-users#1449] [General]
I've successfully integrated fastapi-users into my project and found the setup process straightforward. However, I'm unclear about what happens when a verification request for an email is made. It seems that the library doesn't automatically send an email for verification. Could you please clarify how we should handle sending the verification email ourselves? Specifically, I'm interested in understanding how to generate the email verification link or token within this process.

Thank you for your help

**Comment by hgalytoby** [comment]
You can get the authentication token from here, and then proceed with the email sending logic.

https://fastapi-users.github.io/fastapi-users/latest/configuration/user-manager/#on_after_request_verify

Example of using Background Tasks

https://fastapi.tiangolo.com/tutorial/background-tasks/#background-tasks

https://fastapi-users.github.io/fastapi-users/latest/configuration/user-manager/#create-your-usermanager-class

```py
class UserManager(UUIDIDMixin, BaseUserManager[User, uuid.UUID]):
    reset_password_token_secret = SECRET
    verification_token_secret = SECRET

    def __init__(self, user_db, background_tasks: BackgroundTasks):
        super().__init__(user_db)
        self.background_tasks = background_tasks

    async def on_after_request_verify(
        self, user: User, token: str, request: Optional[Request] = None
    ):
        print(f"Verification requested for user {user.id}. Verification token: {token}")
        await self.send_mail(user=user, token=token)
   
    async def send_mail(self, user: User, token: token):
        ...
        self.background_tasks.add_task(...)


async def get_user_manager(background_tasks: BackgroundTasks, user_db=Depends(get_user_db)):
    yield UserManager(user_db=user_db, background_tasks=background_tasks)
```

There are other celery, arq ... There are many ways to implement.

**Reply by striker561** [reply]
I am familiar with Background Tasks, thanks a lot.
